src/hlstrafe.cpp

```diff
@@ -498,6 +498,33 @@ namespace HLStrafe
 		avec[1] = wishvel[1];
 	}
 
+	void UpdateLookAtViewangle(const PlayerData& player, CurrentState& curState)
+	{
+		if (curState.Parameters.Parameters.LookAt.Entity == 0) {
+			curState.TargetYawLookAtOrigin[0] = curState.Parameters.Parameters.LookAt.X;
+			curState.TargetYawLookAtOrigin[1] = curState.Parameters.Parameters.LookAt.Y;
+			curState.TargetYawLookAtOrigin[2] = curState.Parameters.Parameters.LookAt.Z;
+		} else {
+			curState.TargetYawLookAtOrigin[0] += curState.Parameters.Parameters.LookAt.X;
+			curState.TargetYawLookAtOrigin[1] += curState.Parameters.Parameters.LookAt.Y;
+			curState.TargetYawLookAtOrigin[2] += curState.Parameters.Parameters.LookAt.Z;
+		}
+
+		float difference[3];
+		VecSubtract<float, float, 3>(curState.TargetYawLookAtOrigin, player.Origin, difference);
+		double yaw = Atan2(difference[1], difference[0]) * M_RAD2DEG;
```

Comment:
Something tells me this will break if target origin == player origin. Needs a check

Reply:
Yea, if looking at the player then it is some weird YAW looks behind the player, right at the head, wherever it is supposed to be.

Reply:
I think this should be fine

https://user-images.githubusercontent.com/29143195/210155375-f938de0a-a118-4e6b-82c4-8b05bb01b781.mp4

Reply:
> If a domain error occurs, an implementation-defined value is returned (NaN where supported) 

Bad idea to just let NaNs hang around, better add a check.